### PR TITLE
Dev 3 update account

### PR DESCRIPTION
**Pull request by Four58**

### CODE REVIEW

**Review by Four58 — COMMENTED** [review]
good job, myself

File: src/hooks/imageUpload.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
   const uploadHandler = useCallback((e: ChangeEvent<HTMLInputElement>) => {
     const file = e.target.files?.[0];
     if (file !== undefined) {
-      const base64 = convertToBase64(file);
-      if (typeof base64 === "string") setImageData(base64);
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setImageData(reader.result?.toString() as string);
+      };
+      reader.readAsDataURL(file);
```
Comment (swsd2544) [inline src/hooks/imageUpload.ts:12]: Is `reader.onloadend` guarantees that `reader.result` will have toString method?

File: src/pages/auth/new-user.tsx [hunk 2]
```diff
@@ -50,22 +44,24 @@ const NewUser = () => {
...
     handleSubmit,
     formState: { errors },
-  } = useForm<FormValues>({ resolver });
+  } = useForm<ValidationSchema>({
+    resolver: zodResolver(validationSchema),
+  });
   const router = useRouter();
   const updateUser = api.user.update.useMutation({
     onSuccess() {
       refocusWindow();
```
Comment (swsd2544) [inline src/pages/auth/new-user.tsx:53]: Remove this and the function since we don't need to use this anymore.

**Review by swsd2544 — CHANGES_REQUESTED** [review]
Please fix

File: src/hooks/imageUpload.ts [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
     if (file !== undefined) {
-      const base64 = convertToBase64(file);
-      if (typeof base64 === "string") setImageData(base64);
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        if (reader.result) {
+          setImageData(reader.result?.toString());
+        }
+      };
+      reader.readAsDataURL(file);
```
Comment (swsd2544) [inline src/hooks/imageUpload.ts:14]:
```typescript
const reader = new FileReader();
reader.readAsDataURL(file);
reader.onloadend = () => {
   const base64 = reader.result;
   if (typeof base64 === "string") setImageData(base64); 
};
```

**Review by swsd2544 — CHANGES_REQUESTED** [review]
needs fix

**Review by swsd2544 — APPROVED** [review]

**Pull request merged**